### ISSUE TITLE
ArgumentOutOfRangeException on ToSkBitmapAsync

### ISSUE THREAD

**Issue by Matti-Koopa** [DrJohnMelville/Pdf#41]
(removed)
Code:
```
for (int page = 1; page <= document.TotalPages; page++)
{
  var images = await document.ImagesFromAsync(page).ConfigureAwait(false);
  foreach (var image in images)
  {
	  using var skBitmap = await image.ToSkBitmapAsync().ConfigureAwait(false);
	  (...)
  }
}
```

```
System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'minimumSize')
   at System.IO.Pipelines.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument)
   at System.IO.Pipelines.PipeReader.ReadAtLeastAsync(Int32 minimumSize, CancellationToken cancellationToken)
   at Melville.JBig2.SegmentParsers.SegmentReader.ReadSpanAsync()
   at Melville.JBig2.SegmentParsers.SegmentReader.ReadDataFromAsync()
   at Melville.JBig2.JbigPageReader.ReadSegmentsAsync(SegmentHeaderReader segmentHeadReader)
   at Melville.Pdf.LowLevel.Filters.Jbig2Filter.JbigToPdfAdapter.DecodeOnReadStreamAsync(Stream input, PdfDirectObject parameters)
   at Melville.Pdf.LowLevel.Filters.FilterProcessing.StaticSingleFilter.DecodeAsync(Stream source, PdfDirectObject filter, PdfDirectObject parameter)
   at Melville.Pdf.LowLevel.Filters.FilterProcessing.SinglePredictionFilter.DecodeAsync(Stream source, PdfDirectObject filter, PdfDirectObject parameter)
   at Melville.Pdf.LowLevel.Filters.CryptFilters.CryptSingleFilter.DecodeAsync(Stream source, PdfDirectObject filter, PdfDirectObject parameter)
   at Melville.Pdf.LowLevel.Filters.FilterProcessing.FilterProcessor.DecodeAsync(Stream source, StreamFormat sourceFormat, StreamFormat targetFormat)
   at Melville.Pdf.LowLevel.Filters.FilterProcessing.FilterProcessorBase.StreamInDesiredEncodingAsync(Stream src, StreamFormat sourceFormat, StreamFormat desiredEncoding)
   at Melville.Pdf.LowLevel.Filters.FilterProcessing.FilterProcessorBase.StreamInDesiredEncodingAsync(Stream src, StreamFormat sourceFormat, StreamFormat desiredEncoding)
   at Melville.Pdf.LowLevel.Model.Objects.PdfStream.StreamContentAsync(StreamFormat desiredFormat, IObjectCryptContext encryptor)
   at Melville.Pdf.Model.Renderers.Bitmaps.PdfBitmapWrapper.InnerRenderAsync(BitmapWriter c)
   at Melville.Pdf.SkiaSharp.ConvertPdfBitmap.ToSkBitmapAsync(IPdfBitmap bitmap)
```

**Comment by DrJohnMelville** [comment]
Thanks for the bug report.  I can repro this.

The bug is in the JBIG parser.  The segment 1 length is reported as 0xFFFFFFFF.  The segment cannot be this long, because the file is not that big.  I am going to have to dive into the JBIG spec and see what the expected behavior for this size of a segment is.  The current implementation -- Try to load 0xFFFFFFFF bytes from the source stream into a memory buffer is clearly erroneous.

I'm not even sure if this is a signed or unsigned value but either way this looks for all the world like a sentinel value that needs some special handling.

**Comment by Matti-Koopa** [comment]
Probably some weird scan to PDF programs. I'm getting all those weird edge case with my current dataset. Submitted construction plans and permits from all sort of firms. Huge images. Somewhat interesting, somewhat frustrating. If you want I can share the dataset (few hundered different PDFs) with you. Won't upload in public to GitHub for privacy concerns.

**Comment by DrJohnMelville** [comment]
I think this issue is fixed in commit 3273e80f.  I am going to shortly push a point release to Nuget with the three bugs you identified so far (plus a couple that I caught) fixed.

Please note that the document referenced includes a signature and a date that render differently in Melville.PDF than in acrobat.  The signature and date are done as annotations, which are currently unsupported in Melville.PDF.  (I may do them in the near future.)

I would be interested in your document library, as right now for my robustness goals I need to expose this code to a lot of PDFs.  I could use that conribution.  Alternatively, if you can clone the repository and build it, look at the FuzzTest program.  Its name is more ambitious than what it really does.  If you run FuzzTest with a directory as a parameter FuzzTest will search that directory and all its children for files with a .pdf extension.  It will try to render every page of every pdf file it finds and then it will report and exceptions that get thrown in the rendering process.

So your choice.  You can send me the files at johnmelville@gmail.com or you can just run FuzzTest and open issues on the failures.  Either was I would be most grateful.
